ckb_dlfcn.h

```diff
@@ -259,7 +259,13 @@ int ckb_dlopen2(const uint8_t *dep_cell_hash, uint8_t hash_type,
         if (ph->p_offset < prepad) {
           return ERROR_ELF_NOT_ALIGNED;
         }
-        ret = _ckb_load_cell_code(addr2, memsz, ph->p_offset - prepad, ph->p_filesz,
+        uint64_t offset = ph->p_offset;
```

Comment:
I wasn't following the logic here.

The original code does have a bug indeed, I also noticed it when I was tweaking other code. I believe the following code can already work:

```
ret = _ckb_load_cell_code(addr2, memsz, ph->p_offset - prepad, ph->p_filesz + prepad, index, CKB_SOURCE_CELL_DEP);
```

I wonder if there's anything else needed.